File: aat-lib/src/main/java/ch/bailu/aat_lib/service/location/MissingTrigger.java
package ch.bailu.aat_lib.service.location;

import ch.bailu.aat_lib.logger.AppLog;
import ch.bailu.aat_lib.preferences.StorageInterface;
import ch.bailu.aat_lib.preferences.presets.SolidMissingTrigger;

public final class MissingTrigger extends LocationStackChainedItem {
    private int triggerMillis=15000;
    private long stamp = System.currentTimeMillis();

    public MissingTrigger(LocationStackItem n) {
        super(n);
    }


    @Override
    public void passLocation(LocationInformation location) {
        AppLog.d(this, "pass location");

        stamp=location.getTimeStamp();
        super.passLocation(location);


    }

    public boolean isMissingUpdates() {
        return System.currentTimeMillis()-stamp > triggerMillis;
    }

    @Override
    public void onPreferencesChanged(StorageInterface storage, String key, int presetIndex) {
        triggerMillis=new SolidMissingTrigger(storage, presetIndex).getTriggerMillis();
    }


}

File: aat-lib/src/main/java/ch/bailu/aat_lib/service/location/MissingTrigger.kt
package ch.bailu.aat_lib.service.location

import ch.bailu.aat_lib.preferences.StorageInterface
import ch.bailu.aat_lib.preferences.presets.SolidMissingTrigger
import javax.annotation.Nonnull

class MissingTrigger(next: LocationStackItem) : LocationStackChainedItem(next) {
    private var triggerMillis = 15000
    private var stamp = System.currentTimeMillis()

    override fun passLocation(@Nonnull location: LocationInformation) {
        stamp = location.getTimeStamp()
        super.passLocation(location)
    }

    val isMissingUpdates: Boolean
        get() = System.currentTimeMillis() - stamp > triggerMillis

    override fun onPreferencesChanged(storage: StorageInterface, key: String, presetIndex: Int) {
        triggerMillis = SolidMissingTrigger(storage, presetIndex).triggerMillis
    }
}
